src/app/login/container.tsx

```diff
@@ -66,18 +65,60 @@ export default class Login extends React.PureComponent<ComponentProps, Component
         isFirstInstallation: isEmpty(userSettings),
       })
 
-      this.startDaemon()
+      await this.startDaemon()
+
+      // Now that we have started the wallet daemon,
+      // we need to start the first login process if this is a first installation.
+      if (this.state.isFirstInstallation) {
+        this.startFirstInstallation()
+
+        return
+      }
+
+      const walletStartData: WalletStartDataHard =
+        pick<UserSettings, 'addresses' | 'masterNodeAddress' | 'randomAddresses'>(
```

Comment:
Do we actually need this very complicated Type on a third party library?

Reply:
Yes it's important to have a full typing so that if any change is made and happens to be wrong, you would immediately see that it's wrong.

This typing is verbose I agree (that's the issue with heavy typings in general), however it's kind of easy to understand isn't it ?